Products/ZenUI3/browser/resources/js/zenoss/itinfrastructure.js

```diff
@@ -1408,12 +1408,45 @@ Ext.define("Zenoss.InfraDetailNav", {
     }
 });
 
+var truncate = function(cmp_name, width, uid) {
+    var path = uid.replace('/zport/dmd/', ''),
+        textwidth = Ext.util.TextMetrics.measure(cmp_name, path).width;
+    if (textwidth < width) {
+        return path;
+    } else {
+        var path_array = path.split('/'),
+            device_class = path_array[path_array.length-1];
+
+        // check if DeviceClass's name length > field's length
+        if (Ext.util.TextMetrics.measure(cmp_name, ('../'+device_class)).width < width) {
+            path_array[path_array.length-2] = '..';
+            while (path_array.length >= 3 && Ext.util.TextMetrics.measure(cmp_name, path_array.join('/')).width >= width) {
+                path_array.splice(path_array.length-3, 1);
+            }
+            return path_array.join('/');
+        } else {
+            // automatic ellipsis by CSS
+            return '../'+device_class;
+        }
+    }
+}
 
 var device_grid = Ext.create('Zenoss.DeviceGridPanel', {
     ddGroup: 'devicegriddd',
     id: 'device_grid',
     multiSelect: true,
-    title: _t('/'),
+    header: {
+        titlePosition: 0,
+        items: [{
+                   xtype:'displayfield',
+                   value: _t('/'),
+                   width: 'auto',
+                   cls: 'x-device-panel-header',
+                   id: 'device_panel_header',
+                   allowBlank : true,
+                   readOnly: true
+               }]
```

Comment:
Please deindent this by two levels.  It doesn't need to be pushed so far to the right.  E.g.
```
    items: [{
        xtype:'displayfield',
        value: _t('/'),
        width: 'auto',
        cls: 'x-device-panel-header',
        id: 'device_panel_header',
        allowBlank : true,
        readOnly: true
    }]
```